opensrp-chw/src/main/java/org/smartregister/chw/dao/FamilyKitDao.java

```diff
@@ -3,23 +3,24 @@
 import org.smartregister.chw.anc.domain.VisitDetail;
 import org.smartregister.dao.AbstractDao;
 
-import java.util.Date;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 public class FamilyKitDao extends AbstractDao {
 
     public static long getLastFamilyKitDate(String familyBaseEntityID) {
-        String sql = "select eventDate from event where eventType = 'Family Kit' and " +
-                "baseEntityId = '" + familyBaseEntityID + "' order by eventDate desc limit 1";
+        String sql = "select CASE WHEN created_at <= visit_date THEN created_at ELSE visit_date END family_kit_date from visits where visit_type = 'Family Kit' and " +
```

Comment:
There are a few helper methods to read long values and a single value query.. e.g, you can simplify this to 

```
        String sql = "select CASE WHEN created_at <= visit_date THEN created_at ELSE visit_date END family_kit_date from visits where visit_type = 'Family Kit' and " +
                "base_entity_id = '" + familyBaseEntityID + "' order by created_at desc limit 1";

        DataMap<Long> dataMap = c -> getCursorLongValue(c, "family_kit_date");

        Long res = AbstractDao.readSingleValue(sql, dataMap);
        return res == null ? 0 : res;
```